bndtools.core/src/bndtools/editor/project/RunRequirementsPart.java

```diff
@@ -269,11 +271,7 @@ private void doAddBundle() {
                     Requirement req = new CapReqBuilder(IdentityNamespace.IDENTITY_NAMESPACE).addDirective(Namespace.REQUIREMENT_FILTER_DIRECTIVE, filter.toString()).buildSyntheticRequirement();
                     adding.add(req);
                 }
-                if (!adding.isEmpty()) {
```

Comment:
Here you remove the empty check before calling the new method. Below you keep the empty check. I would assume these should be consistent.